src/dart-2.0.md

```diff
@@ -0,0 +1,95 @@
+---
+layout: default
+permalink: /dart-2.0
+title: Dart 2.0 Updates
+description: How Dart 2.0 is different than 1.x, and how you can prepare.
+---
+
+This page has information about changes that are coming in Dart 2.0.
+Some changes may not apply to your target platform.
+
+* [Changes in Dart 2.0](#changes-in-dart-2.0)
+  * [Web](#web)
+  * [Mobile (Flutter)](#mobile-flutter)
+  * [Servers and command line](#servers-and-command-line)
+* [Removed from Dart 2.0](#removed-from-dart-2.0)
+
+---
+
+You can implement these changes now,
+but keep watching this page for updates. We'll let you know what you can
+do now, and what needs to wait for Dart 2.0.
+
+## Changes in Dart 2.0
+
+Dart 1.x is an optionally typed language, but Dart 2.0 requires static types.
```

Comment:
I agree. It is very confusing (even if it is somewhat technically accurate) to say that D2 "requires static types". What is confusing is that D2 does not require the user to specify these static types if they can be determined automatically through type inference. As Seth points out, "var x = 1;" is valid because D2 determines that x is an integer and (statically) types it as such. Languages like Java and Pascal require types to be explicit and require more work by the programmer, while D2 does not.

Also confusing is the fact that in D2 the type "dynamic" is a static type. The user can declare a type (or it can be inferred) as dynamic. That declares (statically) that the type can be dynamic, so it can actually be any type at runtime. Likewise, in D2 a num can be either an int or a double at runtime. Again, a static type that behaves like a dynamic type at runtime.

The bottom line is that we need to make it clear that while D2 is statically typed, the provision of type inference and types (like dynamic and num) that behave in dynamic ways at runtime, takes much of the burden of specifying types away from the developer. You could even say that D2 offers the advantages of a strongly typed language, while only requiring slightly more work on the part of the programmer (compared to D1).

Even further, D2 has a sound type system, which is even better than strongly typed (a statically typed language is not necessarily sound, One famous example is the C language, which didn't distinguish a pointer from an int or an array, among other things).

I can help wordsmith this if you want.

BTW, this is a short but good overview of typing -- http://blog.steveklabnik.com/posts/2010-07-17-what-to-know-before-debating-type-systems